v3/newrelic/sqlparse/sqlparse.go

```diff
@@ -55,15 +176,32 @@ var (
 // Ability to correctly parse queries for other SQL databases is not
 // guaranteed.
 func ParseQuery(segment *newrelic.DatastoreSegment, query string) {
-	s := cCommentRegex.ReplaceAllString(query, "")
-	s = lineCommentRegex.ReplaceAllString(s, "")
-	s = sqlPrefixRegex.ReplaceAllString(s, "")
-	op := strings.ToLower(firstWordRegex.FindString(s))
-	if rg, ok := sqlOperations[op]; ok {
+	sql := sqlTokenizer{query: query}
+	op := strings.ToLower(removeAllComments(sql.nextWord()))
+	if tablePrefix, ok := operations[op]; ok {
 		segment.Operation = op
-		if nil != rg {
-			if m := rg.FindStringSubmatch(s); len(m) > 1 {
-				segment.Collection = extractTable(m[1])
+		if tablePrefix != "" {
+			for {
+				word := sql.nextWord()
+				if word == "" {
+					break
+				}
+				if strings.ToLower(word) == tablePrefix {
+					table := sql.nextToken()
+					segment.Collection = extractTable(removeAllComments(table))
```

Comment:
Won't the call to `nextToken` result in a token free of comments?

Reply:
.... Answering my own question. I think a comment could be between the `[` and the `]`.